### PR TITLE
15 refactor prometheus targets

### LINKED ISSUE #15: Refactor prometheus_tdp_components_targets variable and Prometheus targets generation logic
The `prometheus_tdp_components_targets` variable has to be refactored to support being overrided/enriched by users. This can be done by using a dict instead of an array.

https://github.com/TOSIT-IO/tdp-observability/blob/415d32ec7ddf48107af3d924173e3bd3d17c4079/tdp_vars_defaults/prometheus/prometheus.yml#L32-L36

Also, the Prometheus targets generation should skip components that are not installed on the cluster to avoid `DOWN` alerts.

### PR DESCRIPTION
**Pull request by GuillaumeHold**
#### Which issue(s) this PR fixes

Fixes #15 .

#### Additional comments

Additional targets can be added with the variable `prometheus_tdp_custom_targets`
Targets are filtered by the groups defined in the inventory.

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp_vars_defaults/prometheus/prometheus.yml [hunk 1]
```diff
@@ -30,87 +30,109 @@ prometheus_web_config:
...
     port: "{{ exporter_ranger_kms_http_port }}"
     service: Ranger
-  - component: knox_gateway
+  knox_gateway:
+    component: knox_gateway
     group: knox
     port: "{{ exporter_knox_gateway_http_port }}"
     service: Knox
 
+prometheus_tdp_custom_targets: {}
```
Comment (rpignolet) [inline tdp_vars_defaults/prometheus/prometheus.yml:134]: As I understand it, this variable allows a user to add targets in addition to the automatically generated TDP targets? If so, in `tdp-collection-prerequisites` I used specific names for these items. I used `generated` for items generated by TDP (https://github.com/TOSIT-IO/tdp-collection-prerequisites/blob/ff7e1c1d49b9894fdd3801b349284c085d6eafda/roles/postgresql/defaults/main.yml#L52) and `additional` for items the user wants added (https://github.com/TOSIT-IO/tdp-collection-prerequisites/blob/ff7e1c1d49b9894fdd3801b349284c085d6eafda/roles/postgresql/defaults/main.yml#L69). Then I used a third variable which concatenates the first two variables (https://github.com/TOSIT-IO/tdp-collection-prerequisites/blob/ff7e1c1d49b9894fdd3801b349284c085d6eafda/roles/postgresql/defaults/main.yml#L71). This allows the user to add elements by setting the `additional` variable but if he wishes to remove the generated elements he can by setting the third variable which will remove the concatenation.

File: roles/prometheus/server/tasks/config.yml [hunk 2]
```diff
@@ -28,29 +28,31 @@
...
+    port: "{{ item.value.port }}"
     labels:
-      component: "{{ item.component }}"
+      component: "{{ item.value.component }}"
       cluster: "{{ cluster_name }}"
-      service: "{{ item.service }}"
-  loop: "{{ prometheus_tdp_components_targets }}"
+      service: "{{ item.value.service }}"
+  when: item.value.group in groups.keys()
+  loop: "{{ prometheus_tdp_generated_targets | combine(prometheus_tdp_additional_targets) | dict2items)}}"
```
Comment (rpignolet) [inline roles/prometheus/server/tasks/config.yml:40]: Can you move the `combine` to `tdp_vars` by setting a variable `prometheus_tdp_components_targets` like this:
```
prometheus_tdp_components_targets: {{ prometheus_tdp_generated_targets | combine(prometheus_tdp_additional_targets) }}
```
And use this variable:
```suggestion
  loop: "{{ prometheus_tdp_components_targets | dict2items)}}"
```
It allows the user to disable TDP generated targets.

Comment (rpignolet) [reply roles/prometheus/server/tasks/config.yml:40]: Wait we should talk about that, I'm not sure how it will be merge.

File: playbooks/files [hunk 3]
```diff
@@ -0,0 +1 @@
+../../../../files
```
Comment (rpignolet) [inline playbooks/files:1]: Why adding this symlink ? We do not have it in other collection.

**Review by rpignolet — CHANGES_REQUESTED** [review]

File: playbooks/files (hunk 3 above)
Comment (GuillaumeHold) [reply playbooks/files:1]: it was generated by the setup.sh. I will remove it from the commit.

**Review by rpignolet — APPROVED** [review]

**Pull request merged**